From 510c5a484d2464721021ff06d210ef8756fdd0c9 Mon Sep 17 00:00:00 2001
Message-Id: <510c5a484d2464721021ff06d210ef8756fdd0c9.1426836508.git.chang-joon.lee@intel.com>
From: vkorjani <vikas.korjani@intel.com>
Date: Mon, 2 Mar 2015 14:59:24 +0530
Subject: [PATCH 1/2] REVERTME [VPG]: Fixing the DSI enable calls sequence

This patch makes the DSI encoder->enable() to be called after pipe
enable like done for any other encoder, no exceptions needed.

REVERTME: This change is available in upstream as part of a patch
which cannot be taken directly. we can revert this patch in next
forklift.
The reference patch in upstream is:
Upstream-Repo: http://anongit.freedesktop.org/git/drm-intel
commit         5004945f1d6c0282c0288afa89ad85d7f2bea4d5
Author:        Jani Nikula <jani.nikula@intel.com>
drm/i915:      move encoder->enable callback later in VLV crtc enable

Issue: GMINL-2254
Change-Id: I3254216f92a4cb8f6319d8053e82dea4a62891a6
Signed-off-by: vkorjani <vikas.korjani@intel.com>
Signed-off-by: Yogesh Mohan Marimuthu <yogesh.mohan.marimuthu@intel.com>
---
 drivers/gpu/drm/i915/intel_display.c |   18 +++---------------
 1 file changed, 3 insertions(+), 15 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index 52155f4..cab0c6d 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -5401,19 +5401,10 @@ static void valleyview_crtc_enable(struct drm_crtc *crtc)
 			vlv_enable_pll(intel_crtc);
 	}
 
-	for_each_encoder_on_crtc(dev, crtc, encoder) {
+	for_each_encoder_on_crtc(dev, crtc, encoder)
 		if (encoder->pre_enable)
 			encoder->pre_enable(encoder);
 
-		if (encoder->type == INTEL_OUTPUT_DSI) {
-			/*
-			 * For DSI recommended to enable
-			 * PORT before plane and pipe
-			 */
-			encoder->enable(encoder);
-		}
-	}
-
 	i9xx_pfit_enable(intel_crtc);
 
 	intel_crtc_load_lut(crtc);
@@ -5430,11 +5421,8 @@ static void valleyview_crtc_enable(struct drm_crtc *crtc)
 
 	intel_enable_pipe(intel_crtc);
 
-	for_each_encoder_on_crtc(dev, crtc, encoder) {
-		if (encoder->type != INTEL_OUTPUT_DSI)
-			/* For DSI already enabled above */
-			encoder->enable(encoder);
-	}
+	for_each_encoder_on_crtc(dev, crtc, encoder)
+		encoder->enable(encoder);
 
 	intel_crtc_enable_planes(crtc);
 	intel_update_drrs(dev);
-- 
1.7.9.5

